tests/Integration/PoolIntegrationTest.php

```diff
@@ -42,6 +42,48 @@ protected function setUp(): void
         parent::setUp();
     }
 
+    public static function invalidKeys()
+    {
+        $bigKey = '';
+        for ($i = 102500; $i > 0; $i--) {
+            $bigKey .= 'a';
+        }
+
+        return array_merge(
+            self::invalidArrayKeys(),
+            [
+                [''],
+                [$bigKey]
+            ]
+        );
+    }
+
+    /**
+     * meh
+     *
+     * @return array
+     */
+    public static function invalidArrayKeys()
+    {
+        return [
+            [''],
+            [''],
+            [''],
```

Comment:
yup those kinda suck but i've only 2 restricted key scenario when keys are forced into strings (which is the case thanks to PSR-6 v3 from `psr/cache` repository

Reply:
yes rework all this